src/main/java/com/brsanthu/googleanalytics/internal/GoogleAnalyticsImpl.java

```diff
@@ -110,33 +108,27 @@ public DefaultRequest getDefaultRequest() {
     }
 
     @Override
-    public Future<GoogleAnalyticsResponse> postAsync(GoogleAnalyticsRequest<?> request) {
-        if (!config.isEnabled() || !inSample()) {
-            return null;
+    public GoogleAnalyticsResponse post(GoogleAnalyticsRequest<?> request) {
+        try {
+            return postAsync(request).get();
+        } catch (InterruptedException | ExecutionException e) {
+            throw new RuntimeException(e);
         }
-
-        return executor.submit(() -> post(request));
     }
 
     @Override
-    public GoogleAnalyticsResponse post(GoogleAnalyticsRequest<?> gaReq) {
+    public CompletableFuture<GoogleAnalyticsResponse> postAsync(GoogleAnalyticsRequest<?> gaReq) {
```

Comment:
there was an binary incompatible change here from `java.util.concurrent.Future` to `java.util.concurrent.CompletableFuture`